Sources/TSCUtility/FSWatch.swift

```diff
@@ -428,55 +428,74 @@ public final class Inotify {
 // FIXME: <rdar://problem/45794219> Swift should provide shims for FD_ macros
 
 private func FD_ZERO(_ set: inout fd_set) {
+      #if os(Android)
+	set.fds_bits = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+      #else
 	set.__fds_bits = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
+      #endif
 }
 
 private func FD_SET(_ fd: Int32, _ set: inout fd_set) {
     let intOffset = Int(fd / 16)
     let bitOffset = Int(fd % 16)
-    let mask = 1 << bitOffset
+    let mask: UInt = 1 << bitOffset
+  #if os(Android)
+    var fd_bits = set.fds_bits
+  #else
+    var fd_bits = set.__fds_bits
+  #endif
     switch intOffset {
-        case 0: set.__fds_bits.0 = set.__fds_bits.0 | mask
-        case 1: set.__fds_bits.1 = set.__fds_bits.1 | mask
-        case 2: set.__fds_bits.2 = set.__fds_bits.2 | mask
-        case 3: set.__fds_bits.3 = set.__fds_bits.3 | mask
-        case 4: set.__fds_bits.4 = set.__fds_bits.4 | mask
-        case 5: set.__fds_bits.5 = set.__fds_bits.5 | mask
-        case 6: set.__fds_bits.6 = set.__fds_bits.6 | mask
-        case 7: set.__fds_bits.7 = set.__fds_bits.7 | mask
-        case 8: set.__fds_bits.8 = set.__fds_bits.8 | mask
-        case 9: set.__fds_bits.9 = set.__fds_bits.9 | mask
-        case 10: set.__fds_bits.10 = set.__fds_bits.10 | mask
-        case 11: set.__fds_bits.11 = set.__fds_bits.11 | mask
-        case 12: set.__fds_bits.12 = set.__fds_bits.12 | mask
-        case 13: set.__fds_bits.13 = set.__fds_bits.13 | mask
-        case 14: set.__fds_bits.14 = set.__fds_bits.14 | mask
-        case 15: set.__fds_bits.15 = set.__fds_bits.15 | mask
+        case 0: fd_bits.0 = fd_bits.0 | mask
+        case 1: fd_bits.1 = fd_bits.1 | mask
+        case 2: fd_bits.2 = fd_bits.2 | mask
+        case 3: fd_bits.3 = fd_bits.3 | mask
+        case 4: fd_bits.4 = fd_bits.4 | mask
+        case 5: fd_bits.5 = fd_bits.5 | mask
+        case 6: fd_bits.6 = fd_bits.6 | mask
+        case 7: fd_bits.7 = fd_bits.7 | mask
+        case 8: fd_bits.8 = fd_bits.8 | mask
+        case 9: fd_bits.9 = fd_bits.9 | mask
+        case 10: fd_bits.10 = fd_bits.10 | mask
+        case 11: fd_bits.11 = fd_bits.11 | mask
+        case 12: fd_bits.12 = fd_bits.12 | mask
+        case 13: fd_bits.13 = fd_bits.13 | mask
+        case 14: fd_bits.14 = fd_bits.14 | mask
+        case 15: fd_bits.15 = fd_bits.15 | mask
         default: break
     }
+  #if os(Android)
+    set.fds_bits = fd_bits
+  #else
+    set.__fds_bits = fd_bits
+  #endif
```

Comment:
It'd be nice if I could assign some kind of compile-time symbol reference to each platform's variable name, so that this runtime copying isn't needed.